opentelemetry-otlp/src/lib.rs

```diff
@@ -264,6 +279,27 @@ impl OtlpPipelineBuilder {
         self
     }
 
+    /// Set the trace provider configuration from the given environment variables.
+    ///
+    /// If the value in environment variables is illegal, will fall back to use default value.
+    pub fn with_env(mut self) -> Self {
```

Comment:
I think in other crates we use `from_env` for functions like this.

Reply:
I wanted to use `from_env` but clippy warns about that:
```
warning: methods called `from_*` usually take no self; consider choosing a less ambiguous name
```

Reply:
I think `with_` is actually more idiomatic for this sort of thing.